clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -354,6 +354,21 @@ class GlobalVisitor : public RecursiveASTVisitor<GlobalVisitor> {
     return true;
   }
 
+  bool VisitInitListExpr(InitListExpr *E){
+    if (E->getType()->isStructureType()) {
+      const RecordDecl *Definition =
+          E->getType()->getAsStructureType()->getDecl()->getDefinition();
+
+      unsigned int initIdx = 0;
+      const auto fields = Definition->fields();
+      for (auto it = fields.begin(); initIdx < E->getNumInits() && it != fields.end(); initIdx++, it++) {
+        Expr *InitExpr = E->getInit(initIdx);
+        CB.constrainLocalAssign(nullptr, *it, InitExpr);
+      }
+    }
+    return true;
+  }
+
```

Comment:
This change seems to have nothing to do with `CompoundLIteral`s? I.e., it's just for `struct`s right? If so, this should be a separate commit/PR. Don't want to entangle lots of orthogonal changes; just makes them harder to evaluate.

Reply:
It's related to compound literals in that compound literals contain init lists, so this change needed to happen for the rest of the changes to make sense. That said, it can be easily split out to a separate PR.